### ISSUE TITLE
Not treating `@staticmethod`s as static within an `abc.ABCMeta` (sub)class

### ISSUE THREAD

**Issue by trevenrawr** [charliermarsh/ruff#2634]
While running `ruff 0.0.243` on some code, I had a confusing linting output, which can be reproduced with the following snippet:

```python
from abc import ABCMeta


class MetaClass(ABCMeta):
    @staticmethod
    def i_dont_get_cls_param(not_cls) -> bool:
        return False
```

Despite the method being noted as static, `ruff` is complaining that it should take a `cls` parameter:
```
$ ruff ruff.py
ruff.py:6:30: N804 First argument of a class method should be named `cls`
Found 1 error.
```

Based on the usage of this function in other code, it is indeed static, so this seems like it's some sort of precedence issue between treating all methods of `ABCMeta` as classmethods and the `@staticmethod` decorator.

P.S. Thanks for such a blazingly fast linter and sorter!

**Comment by charliermarsh** [comment]
Thanks for filing! I'll take a look.
